### DISCUSSION TITLE
How would I create breadcrumbs in Next 13?

### DISCUSSION THREAD

**Discussion by joedawson** [vercel/next.js#44701] [Help]
Hey,

My knowledge of Next / React is limited at present so please excuse my ignorance - I am wondering how I would go about creating breadcrumbs in a Next 13 app.

I have built a forum that I would like my breadcrumbs to look something like this from a thread:

```
Home > Forums > [Forum Name] > [Thread Name]
```

How would I create a Breadcrumb component, that preferably sits in my RootLayout, that I can then specify the crumb trail regardless of what route I am on?

For example, let's say I am in my `app/forums/[forum]/page.jsx` and I want my crumb trail to look like this:

```
Home > Forums > [Forum Name]
```

If my page.jsx in `app/forums/[forum]` looked something like this:

```jsx
export default async function ForumPage({ params }) {
  const forum = {} // fetched via params.forum

  const crumbs = [
    { name: 'Forums', href: '/forums' },
    { name: forum.name, href: `/forums/${forum.slug}` },
  ]
}
```

What is the approach to pass in this example, `crumbs`, back up to the RootLayout so I can then map over the crumbs to render?

Or is there a better way to do this?

Thanks in advance!

**Comment by GabenGar** [comment]
Create a separate component devoted to breadcrumbs, run a massive boilerplate `if...else` logic inside it and render in the root layout.
Or write a decorator which would return a layout component with breadcrumbs being defined according to callback.
